src/auto-evo/AutoEvoRun.cs

```diff
@@ -6,9 +6,10 @@
 using System.Threading.Tasks;
 using AutoEvo;
 using Godot;
-using Xoshiro.PRNG64;
 using Thread = System.Threading.Thread;
 
+// using Xoshiro.PRNG64;
```

Comment:
Commented out stuff should be cleaned up if this whole miche feature is completed and it seems like this is no longer required.

Reply:
I'm not sure how / when Xoshiro is supposed to bu handled compared to normal random. I just commented it out as something to come back too when I figured out how it should be used.

Reply:
It should basically always be used instead of normal random. The reason the random is at the top level is so that it can be passed down to anything needing randomness in the auto-evo, and we'd hopefully in the future have fully deterministic auto-evo if given the same random seed and starting conditions. That would be really helpful in debugging any auto-evo problems (for players the results would still be random as the game uses the current time as the random seed when starting a new auto-evo run).

Reply:
Should I go through and replace any new randoms with that or should I leave that for a future issue? Do you have any suggestions on determinism with threads?

Reply:
In .NET 8 the default system random actually defaults to using Xoshiro, but the seeds obviously need to be ferried through the method call sequences otherwise the randoms will not provide a consistent sequence. So I recommend that every place in auto-evo that creates a new random takes in a Random parameter that is used to generate a seed for the new random number generator. That will take care of basically everything that matters. Xoshiro randoms absolutely have to be stored in variables of Xoshiro type *for saving to work* but as auto-evo state is not saved in save files, that is not absolutely necessary.